module-2/backend/week_four.md

```diff
@@ -10,14 +10,29 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+A small piece of data sent to a user and stored on their client as to save any type of information about their 'session' as HTTP is stateless
 * What’s the difference between a session and a cookie?
+A session is stored on the server whereas a cookie is stored in the browser. This makes sessions much more (but still not totally) secure than cookies.
 * What’s a flash and when do you want to use flashes?
+Flash is a hash that stores messages to be displayed to the user depending on actions they take- conformation messages and failure messages.
 * Why do people say “HTTP is stateless”?
+HTTP is stateless because it does not store any information- this is what cookies and sessions are required for- to save any state information that belongs to users.
 * What’s authentication? Explain.
+Authentication is checking a users credentials from a login or other 'specific visit'. It confirms the user is who they say they are with some kind of check.
 * What’s the difference between authentication and authorization?
+Authentication is a one-time log in where authorization is something that persists and checks everytime a user tries to navigate around a website.
```

Comment:
Authentication doesn't have to be a one time thing. Storing a `user_id` in a session would allow us to check to see that a user is who they say they are over time. The main difference is that authorization deals with how we determine if a user is allowed to see some portion of the site.